### DISCUSSION TITLE
What changes have been made to the update from January, it seems that the training effect has become worse

### DISCUSSION THREAD

**Discussion by DuskXi** [d8ahazard/sd_dreambooth_extension#892] [Q&A]
I've been switching versions repeatedly lately, but I've noticed that the versions of sd_dreambooth_extension updated after January are basically having can't training a excellent model as well as the previous versions were trained.

What changes have been made to cause the same amount of data, which used to take only 10,000steps for me to reach the goal and 15,000+steps to overfit, to now take more than 30,000 steps or even 50,000 or 70,000 steps, with the consequent problem that the line between overfitting and underfitting has become blurred.

The next problem is that the step control has been changed to epoch control, and the number of checkpoints saved is not beautiful, but the strangest thing is the above, what exactly has been changed in each training step to cause such a big gap.

My experience of the December 2022 version of sd_dreambooth_extension does not work at all in the current version.

And I have tried to train the model almost 8 times, trying to modify various values, but the results have not been as good as before.

My personal preference is to use the newer stuff rather than keep the old version, so I'm asking if you have any ideas.

**Comment by Renaldas111** [comment]
I am only able to succesfully run version of ~November, current version runs, but learns absolutely nothing, I receive random images during the training preview and after trying to generate outputs.

**Reply by Kahldris** [reply]
You're not alone - https://github.com/d8ahazard/sd_dreambooth_extension/issues/897

I was training successfully a week or so ago though so I don't know where/what broke. No errors, just learns nothing.

**Reply by ye7iaserag** [reply]
If you are using xformers: 0.0.16rc425 downgrade to 0.0.14.dev0 and try again

**Reply by DuskXi** [reply]
> If you are using xformers: 0.0.16rc425 downgrade to 0.0.14.dev0 and try again

it works

**Reply by ye7iaserag** [reply]
I also tested 0.0.17-dev and it's working well

**Comment by FugueSegue** [comment]
I know the troubles you are facing. But I can assure you there is a solution.

At the end of December or the beginning of January, there was an incompatibility issue that arose between the webui and the extension. It was found and addressed. And then almost immediately after that the webui updated something else that made the extension completely incompatible. I considered rolling back my copy of the webui and the extension to earlier commits.

A few days ago I made a completely fresh install of the webui and the extension. This was my general procedure. Keep in mind that I also installed a fresh copy of Windows 10 before I began.

- Installed Visual Studio, Community version without any extra options.
- Installed CUDA Toolkit 11.6.
- Installed Python 3.10.6.
- Installed Git.
- Cloned the webui and copied the SD 1.5 checkpoint into proper directory.
- Launched the webui via the bat file and let everything install as usual. Then I quit the program.
- Added --xformers to the command line argument in the bat file.
- Launched webui again, letting xformers install.
- Installed the Dreambooth extension through the webui's Extensions tab. Close the program.

At this point, I fixed THE PROBLEM. It's really easy.
(For more information, see this informative video: https://youtu.be/O01BrQwOd-Q )

Go to the stable-diffusion-webui-master\venv\Scripts directory, open a command window, and enter these commands:

pip uninstall torch torchvision
pip uninstall torchaudio
pip uninstall xformers
pip install torch torchvision --extra-index-url https://download.pytorch.org/whl/cu116
pip install -U -I --no-deps https://github.com/C43H66N12O12S2/stable-diffusion-webui/releases/download/torch13/xformers-0.0.14.dev0-cp310-cp310-win_amd64.whl

Be sure to IGNORE any strange warnings that those commands might produce. I believe if you do this with the most recent commit of the webui and the most recent commit, everything should be fine.

I have been training Dreambooth normally for several days now.

**Reply by ye7iaserag** [reply]
I'm not sure I understand what you are proposing here, is it a full reinstall? because I run in a docker image and I've rebuilt it multiple times without caches so it should be the same as you recommended yet it still didn't work.
Also there is a new xformers version that I believe is compatible with webui and shall give a speed boost too (0.0.16rc425) ppl have been using it successfully with InvokeUI for a week or two now

**Reply by ye7iaserag** [reply]
After checking I think you are on to something, xformers: 0.0.16rc425 seems to be the common thing here and it might be the reason why we all are having this problem

**Reply by DuskXi** [reply]
The webui after rollback will automatically select xformers 0.0.14

**Comment by cerega66** [comment]
I faced the same problem and I’ll say for sure that the fault is not on the versions of the libraries, since I just change the folder with the extension between the latest and the previous one which works and I didn’t have successful training on the latest version, but returning the folder with the old one gives a positive effect.
